dsd/constraints.py

```diff
@@ -616,17 +652,45 @@ def generate_sequence(self, rng: np.random.Generator) -> str:
 
         :param rng:
             numpy random number generator to use. To use a default, pass :py:data:`np.default_rng`.
+        :param previous_sequence:
+            previously generated sequence to be replaced by a new sequence; optional if no previous
```

Comment:
Instead of "optional", I'd say it's "`None`" if no previous sequence exists.

I'd also say a bit about how the parameter gets used, e.g,. *Used along with :py:data:`DomainPool.hamming_probability` to choose a new sequence "close" to previous_sequence in Hamming distance.*